src/h5web/visualizations/shared/utils.ts

```diff
@@ -37,10 +37,18 @@ export function computeVisSize(
     : { width, height: width / aspectRatio };
 }
 
-export function extendDomain(bareDomain: Domain, extendFactor: number): Domain {
+export function extendDomain(
+  bareDomain: Domain,
+  extendFactor: number,
+  log?: boolean
+): Domain {
   const [min, max] = bareDomain;
-  const extension = (max - min) * extendFactor;
 
+  if (log) {
+    return [min / (1 + extendFactor), max * (1 + extendFactor)];
+  }
+
+  const extension = (max - min) * extendFactor;
   return [min - extension, max + extension];
```

Comment:
The `extendDomain` now accepts a log bool to extend the domain appropriately for log scales. This way, domain bounds never change sign.

This works fine for now but I plan to review this in the future so that we get consistent domain extensions (even for linear scale).